### PR TITLE
fix(otel): fix resource attribute parsing

### LINKED ISSUE #2607: (bug) Tracing: resource attribute parsing is broken (wrong data type)
### Observed behaviour

When setting the standard OTel environment variable `OTEL_RESOURCE_ATTRIBUTES` to a value like `key1=value1,key2=value2`, the go-feature-flag relay crashes:

```json
{
  "level": "fatal",
  "ts": 1730485805.8576112,
  "caller": "relayproxy/main.go:58",
  "msg": "error while reading configuration",
  "error": "1 error(s) decoding:\n\n* 'otel.resource.attributes' expected a map, got 'string'",
  "stacktrace": "main.main\n\t/home/runner/work/go-feature-flag/go-feature-flag/cmd/relayproxy/main.go:58\nruntime.main\n\t/opt/hostedtoolcache/go/1.22.8/x64/src/runtime/proc.go:271"
}
```

Either the type of `OpenTelemetryConfiguration.Resource.Attributes` should be changed to `string` to match the [spec](https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/#general-sdk-configuration), or additional parsing will be necessary.

related: https://github.com/knadh/koanf/issues/273

### Expected Behavior

The `OTEL_RESOURCE_ATTRIBUTES` environment variable should be supported as expected.

### Steps to reproduce

_No response_

### PR DESCRIPTION
**Pull request by hairyhenderson**
## Description

Support for `OTEL_RESOURCE_ATTRIBUTES` was broken as koanf was parsing it, but not as the right type. We need to parse it in a custom way.

## Closes issue(s)

Fixes #2607 

## Checklist
- [x] I have tested this code
- [x] I have added unit test to cover this code
- [x] I have updated the documentation (`README.md` and `/website/docs`)
- [x] I have followed the [contributing guide](CONTRIBUTING.md)

### CODE REVIEW

File: cmd/relayproxy/api/opentelemetry/otel.go [hunk 1]
```diff
@@ -90,9 +91,15 @@ func (o otelErrHandler) Handle(err error) {
 
 var _ otel.ErrorHandler = otelErrHandler(nil)
 
-func initResource(ctx context.Context, serviceName string, version string) (*resource.Resource, error) {
+func initResource(ctx context.Context, serviceName, version string,
+	attribs map[string]string) (*resource.Resource, error) {
+	attrs := make([]attribute.KeyValue, 0, len(attribs))
+	for k, v := range attribs {
+		attrs = append(attrs, attribute.String(k, v))
+	}
```
Comment (thomaspoignant) [inline cmd/relayproxy/api/opentelemetry/otel.go:99]: **praise:** This is a good catch, I think `koanf` is messing up things when it is a complex object. This is way better like this, thank you 🙏.

**Review by thomaspoignant — APPROVED** [review]
@hairyhenderson Thanks again for all your work on the OTEL part this is great and it helps me a LOT 👍 

Sorry for this one, I should have think about this `koanf` weird way of managing complex object.
I'll merge this one and will release it in the next version.

**Comment by hairyhenderson** [comment]
@thomaspoignant 👋 any chance of a patch release soon so I can get this fix rolled out? 🙇‍♂️

**Comment by thomaspoignant** [comment]
@hairyhenderson Yes I will release a version later this week.

**Pull request merged**